### PR TITLE
[#126032645] Taskee bidding system 

### PR DESCRIPTION
**Pull request by Habu-Kagumba**
#### What does this PR do?
- Enable taskee to make bids
- Enable tasker to manage bids
#### What are the relevant pivotal tracker stories?

[#126032645]

### CODE REVIEW

File: app/assets/javascripts/live_notify.js [hunk 1]
```diff
@@ -2,30 +2,30 @@
...
   };
 
   dispatcher.on_open = function (data) {
-    var notificationIndicator = $(".notification-badge span"),
-        notificationWrapper = $(".dashboard-navlinks"),
-        notificationElem = $("<li class='notification-badge'>"),
-        notificationElemChild = $("<span>");
+    var notificationIndicator = $('.notification-badge span'),
+      notificationWrapper = $('.dashboard-navlinks'),
+      notificationElem = $('<li class=\'notification-badge\'>'),
```
Comment (Habu-Kagumba) [inline app/assets/javascripts/live_notify.js]: Linter

File: app/assets/javascripts/task_notifications.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(function () {
```
Comment (Habu-Kagumba) [inline app/assets/javascripts/task_notifications.js]: Changed logic, TO be removed

File: app/assets/stylesheets/styles/_bids.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    color: $primary-color;
+
+    li.active {
+      color: #fff;
+    }
+  }
+}
+
+.pagination {
+  .waves-effect {
```
Comment (Habu-Kagumba) [inline app/assets/stylesheets/styles/_bids.scss:21]: Moves this into .pagination block.

File: app/assets/stylesheets/styles/_bids.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+.taskee_img {
+  padding: 20px;
+  img {
+    border-radius: 50%;
+  }
+}
+
+.taskee_reviews {
```
Comment (Habu-Kagumba) [inline app/assets/stylesheets/styles/_bids.scss]: Un-nest .ratings

File: app/assets/stylesheets/styles/_bids.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  }
+}
+
+.bids {
+  position: relative;
+}
+
+.taskee_img {
+  padding: 20px;
+  img {
```
Comment (Habu-Kagumba) [inline app/assets/stylesheets/styles/_bids.scss:32]: One-liner

File: app/controllers/dashboard/bids_controller.rb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      @bid = Bid.new
+    end
+
+    def create
+      @bid = Bid.new(bid_params)
+      if @bid.save
+        create_bid_notification(@bid)
+        redirect_to [:dashboard, @task], notice: "You have successfully made"\
+          " a bid"
+      else
```
Comment (Habu-Kagumba) [inline app/controllers/dashboard/bids_controller.rb]: eliminate the else

File: app/controllers/dashboard/tasks_controller.rb [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        :min_price,
+        :max_price
+      ).merge(
+        tasker_id: current_user.id,
+        price_range: [params[:task][:min_price], params[:task][:max_price]]
+      )
+    end
+
+    def set_task
+      task = Task.find(params[:id])
```
Comment (Habu-Kagumba) [inline app/controllers/dashboard/tasks_controller.rb:116]: Major refactor

File: app/controllers/notifications_controller.rb [hunk 8]
```diff
@@ -50,4 +54,8 @@ def set_notification_attr(params)
     @message = params[:message]
     @reply_to_sender = params[:reply_to_sender]
   end
+
+  def update_notifiable?
+    !params[:notifiable_attr_to_update].blank?
```
Comment (Habu-Kagumba) [inline app/controllers/notifications_controller.rb]: Use .present?

File: app/views/dashboard/tasks/new.html.erb [hunk 9]
```diff
@@ -4,7 +4,7 @@
     <div class="row">
       <div class="col m8 offset-m2">
         <h3>Create new Task</h3>
-        <%= form_for @task do |f| %>
+        <%= form_for [:dashboard, @task] do |f| %>
           <%= render "partials/error_msg", object:  @task%>
```
Comment (Habu-Kagumba) [inline app/views/dashboard/tasks/new.html.erb:8]: remove space after object:

File: app/assets/javascripts/bids.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+$(function (){
+}())
```
Comment (oeyiowuawi) [inline app/assets/javascripts/bids.js]: remove unused function

File: app/assets/javascripts/task_notifications.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  var notifyBtn = $('.notify-btn')
```
Comment (makinwab) [inline app/assets/javascripts/task_notifications.js]: You shouldn't use abbreviations when naming variables.

File: app/controllers/dashboard/bids_controller.rb [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+    def new
+      @bid = Bid.new
+    end
+
+    def create
+      @bid = Bid.new(bid_params)
+      if @bid.save
+        create_bid_notification(@bid)
+        redirect_to [:dashboard, @task], notice: "You have successfully made"\
```
Comment (makinwab) [inline app/controllers/dashboard/bids_controller.rb]: We can have this notice message in a variable as against breaking this line.

File: app/controllers/dashboard/bids_controller.rb [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+      @task = Task.find(@bid.task_id)
+    end
+
+    def bid_params
+      params.require(:bid).permit(
+        :description,
+        :start_date,
+        :end_date,
+        :price
+      ).merge(
```
Comment (makinwab) [inline app/controllers/dashboard/bids_controller.rb:49]: 💯

File: app/controllers/dashboard/tasks_controller.rb [hunk 14]
```diff
@@ -0,0 +1,173 @@
+module Dashboard
+  class TasksController < ApplicationController
```
Comment (makinwab) [inline app/controllers/dashboard/tasks_controller.rb:2]: **Critical - Code Quality**
This controller is fat. Let's strive to have at most 100 lines per class.

Please let's slim down this class.

**Comment by makinwab** [comment]
Closing this PR because it's no longer relevant to the new direction for the application.

**Pull request closed without merging**